pwiz_tools/BiblioSpec/src/MaxQuantReader.cpp

```diff
@@ -869,7 +869,13 @@ void MaxQuantReader::addLabelModsToVector(vector<SeqMod>& v, const string& rawFi
                                    "raw file '%s'.",
                                    labelingState, labels->labelingStates.size(), rawFile.c_str());
     }
-    const vector<const MaxQuantModification*>& labelMods = labels->labelingStates[labelingState].mods;
+
+    const auto& labelState = labels->labelingStates[labelingState];
+    const auto& labelMods = labelState.mods;
```

Comment:
If the user has an N-terminal amino acid which applies to a particular amino acid, that modification is going to get applied to all of the occurrences of that amino acid, even the ones that are not on the N-terminal, right?

To do this 100% correctly, you would have to separately consider the modifications for all the applicable positions depending on where you are in the peptide sequence.

You would need to duplicate the code in "addModsToVector":
```
            boost::range::insert(v, v.end(), getFixedMods(modSequence[i], i + 1 - modsTotalLength, modsAnywhere));
            if (i == 0)
                boost::range::insert(v, v.end(), getFixedMods(modSequence[i], i + 1 - modsTotalLength, modsAnyNTerm));
            else if (i + 1 == sequenceLength)
                boost::range::insert(v, v.end(), getFixedMods(modSequence[i], i + 1 - modsTotalLength, modsAnyCTerm));

            if (i > 0)
                boost::range::insert(v, v.end(), getFixedMods(modSequence[i], i + 1 - modsTotalLength, modsNotNTerm));
            if (i + 1 < sequenceLength)
                boost::range::insert(v, v.end(), getFixedMods(modSequence[i], i + 1 - modsTotalLength, modsNotCTerm));
```

Is there a reason that it's not worth doing that? Like, are we assuming that people are not going to combine location specificity with amino acids specificity on their label modifications like this?

Reply:
The new code adds labels that are N or C terminal (position) with no site specificity. The ones with site specificity should already be added by the for loop (without any regard to position). Do you have a particular position/site combination you think wouldn't be supported here?

Reply:
Oh, I see what you mean. Yes, the current for loop is potentially too inclusive because it doesn't consider position.